src/filesystem/implementations/local.h

```diff
@@ -60,12 +60,45 @@ class LocalFileSystem : public FileSystem {
   Status MakeTemporaryDirectory(
       std::string dir_path, std::string* temp_dir) override;
   Status DeletePath(const std::string& path) override;
+
+ private:
+  inline std::string getOSValidPath(const std::string& _path);
+  const char* kWindowsLongPathPrefix = "\\\\?\\";
 };
 
+
+//! Converts incoming utf-8 path to an OS valid path
+//!
+//! On Linux there is not much to do but make sure correct slashes are used
+//! On Windows we need to take care of the long paths and handle them correctly
+//! to avoid legacy issues with MAX_PATH
+//!
+//! More details:
+//! https://learn.microsoft.com/en-us/windows/win32/fileio/maximum-file-path-limitation?tabs=registry
+//!
+inline std::string
+LocalFileSystem::getOSValidPath(const std::string& _path)
```

Comment:
```suggestion
LocalFileSystem::getOSValidPath(const std::string& path)
```
Cascading effect where the local `path` variable will need to change to something like `local_path` or `l_path`